src/Authentication/ApiKey.php

```diff
@@ -0,0 +1,379 @@
+<?php
+/**
+ * API key.
+ *
+ * @package SatisPress
+ * @license GPL-2.0-or-later
+ * @since 0.3.0
+ */
+
+declare ( strict_types = 1 );
+
+namespace SatisPress\Authentication;
+
+use function SatisPress\generate_random_string;
+use ArrayAccess;
+use DateTime;
+use DateTimeZone;
+use WP_User;
+use WP_User_Query;
+
+/**
+ * API key class.
+ *
+ * @since 0.3.0
+ */
+final class ApiKey implements ArrayAccess {
+	/**
+	 * Prefix for user meta keys.
+	 *
+	 * @var string
+	 */
+	const META_PREFIX = 'satispress_api_key.';
+
+	/**
+	 * API key length.
+	 *
+	 * @var integer
+	 */
+	const TOKEN_LENGTH = 32;
+
+	/**
+	 * String to prepend to API keys.
+	 *
+	 * @var string
+	 */
+	const TOKEN_PREFIX = '';
+
+	/**
+	 * API key data.
+	 *
+	 * @var array
+	 */
+	protected $data;
+
+	/**
+	 * API key token.
+	 *
+	 * @var string
+	 */
+	protected $token;
+
+	/**
+	 * User associated with the API key.
+	 *
+	 * @var WP_User
+	 */
+	protected $user;
+
+	/**
+	 * Initialize an API key.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param WP_User $user  WordPress user.
+	 * @param string  $token API key token.
+	 * @param array   $data  Additional data associated with the key.
+	 */
+	public function __construct( WP_User $user, string $token, array $data = [] ) {
+		$this->user  = $user;
+		$this->token = $token;
+		$this->data  = $data;
+	}
+
+	/**
+	 * Retrieve the API Key name.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return $this->data['name'] ?? '';
+	}
+
+	/**
+	 * Retrieve the API Key token.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_token(): string {
+		return $this->token;
+	}
+
+	/**
+	 * Retrieve the user associated with the API Key.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return WP_User
+	 */
+	public function get_user(): WP_User {
+		return $this->user;
+	}
+
+	/**
+	 * Retrieve and format a date field.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param string $name   Field name.
+	 * @param string $format Optional. Date format.
+	 * @return mixed
+	 */
+	public function get_date( string $name, string $format = null ) {
+		if ( empty( $this->data[ $name ] ) ) {
+			return '';
+		}
+
+		return $this->format_date( $this->data[ $name ], $format );
+	}
+
+	/**
+	 * Revoke the API key.
+	 *
+	 * @since 0.3.0
+	 */
+	public function revoke() {
+		delete_user_meta(
+			$this->get_user()->ID,
+			$this->get_meta_key( $this->token )
```

Comment:
The call to this method in the `save()` method is done statically, but this instance uses `$this->`. Why the inconsistency?

Reply:
Blindness from staring at it too long 😵 That's why it's nice to have another pair of eyes on it.